app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -9,31 +9,94 @@ module ClassMethods
     def validate_config_info(options)
       config_info = options[:config_info]
 
+      # ------------------------------------------------------------------------
+      # TransformationMapping validation
+      # ------------------------------------------------------------------------
       mapping = if config_info[:transformation_mapping_id]
                   TransformationMapping.find(config_info[:transformation_mapping_id])
                 else
                   config_info[:transformation_mapping]
                 end
-
+      # mapping.invalid? added as a part of validating migration plan
       raise _('Must provide an existing transformation mapping') if mapping.blank?
+      unless mapping.valid?
+        raise _('Transformation mapping was invalid.  ' + mapping.errors.messages)
+      end
 
+      # ------------------------------------------------------------------------
+      # check for openstack as a target for Flavors and SecurityGroup validation
+      # ------------------------------------------------------------------------
+      mapping_item        = mapping.transformation_mapping_items.find_by(:destination_type => "CloudTenant")
+      cloud_tenant        = mapping_item.destination if mapping_item.present?
+      osp_flavors         = cloud_tenant.flavors if cloud_tenant.present?
+      osp_security_groups = cloud_tenant.security_groups if cloud_tenant.present?
+
+      # ------------------------------------------------------------------------
+      # ansible playbooks validation
+      # ------------------------------------------------------------------------
       pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
       post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
 
+      # Add as a part of validating migration plan
+      # check service_type. Filter type=ServiceTemplateAnsiblePlaybook
+      if pre_service_id.present?
+        pre_service = ServiceTemplate.find(pre_service_id)
+        unless pre_service.nil?
+          raise _('Premigration service type MUST be "ServiceTemplateAnsiblePlaybook"') if pre_service.type != "ServiceTemplateAnsiblePlaybook"
```

Comment:
Looks like my comment got lost, but instead of checking for `.type`, this is STI so you can just check with the class like this `ServiceTemplateAnsiblePlaybook.find(pre_service_id)` then you don't need to check the class type